Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8059,6 +8059,10 @@ def parse_date_string(date_string, date_format='%Y-%m-%dT%H:%M:%S'):
             # found timezone - appending it to the date format
             date_format += time_zone[0]
 
+        # Make the fractions shorter less than 6 charactors
+        # e.g. '2022-01-23T12:34:56.123456789+09:00' to '2022-01-23T12:34:56.123456+09:00'
+        date_string = re.sub(r'(.*?\.[0-9]{6})[0-9]*([Zz]|[+-]\S+?)', '\\1\\2', date_string)
```

Comment:
In case there is no suffix of + or - or Z or z like -> `2019-09-17T06:16:39.4040555555555` it fails.
Can't a regex like this give the solution for all cases?
```suggestion
        date_string = re.sub(r'(.*?\.[0-9]{6})[0-9]*', '\\1', date_string)
```

Reply:
@israelpoli OK, I've updated the regex to make it better to handle it.